.github/workflows/deploy.yml

```diff
@@ -0,0 +1,40 @@
+name: Deploy to GitHub Pages
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  deploy:
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: write
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.12
+
+      - name: Install MkDocs & co
+        run: pip install -r requirements.txt
+
+      - name: Build site
+        run: mkdocs build
+
+      - name: Deploy
+        uses: peaceiris/actions-gh-pages@v4
+        if: github.ref == 'refs/heads/main'
+        with:
+          personal_token: ${{ secrets.ACTIONS_DEPLOY_TOKEN }}
+          external_repository: Quansight-Labs/pyfreethreading.github.io
```

Comment:
We're gonna need a new repo under the `Quansight-Labs` org that'll host the generated static assets.

Reply:
We don't I think - it can go on the `gh-pages` branch of this repo. Which is what the `peaceiris/actions-gh-pages` does by default. I believe we should be able to use https://github.com/pypackaging-native/pypackaging-native/blob/main/.github/workflows/deploy.yml pretty much unchanged.

Reply:
That's right, although this is what https://github.com/pypackaging-native/pypackaging-native/blob/main/.github/workflows/deploy.yml does and that's why I used it. It deploys static assets in this repo: https://github.com/pypackaging-native/pypackaging-native.github.io.

Reply:
I removed this and created a `gh-pages` branch on this repo. Maybe it still makes sense to do it in a different one to chose another name for the site? `free-threaded-compatibility.github.io` feels a bit big. I used `pyfreethreading.github.io` in the `mkdocs` config.

Reply:
Ah yes, domain name fun. The default is `quansight-labs.github.io/free-threaded-compatibility` here. It's the GitHub org name rather than the repo name that goes at the front of the URL.

Choosing a custom domain name we don't get from GitHub Pages is a long-term maintenance hassle (more worried about that then about the $ for a domain name), so I'm inclined to stay with the default.

Reply:
Oh right. What we could do is set up an organization, create a repo under there and have the action push to that repo, if we want to. We only have to maintain the org/repo but not a full-fledged domain. Otherwise, we can just go ahead and leave everything as-is.

Reply:
Good idea @lysnikolaou. I think this is actually worth doing. The site is going to be up for a very long time, so best to have a memorable name. And no need to have `quansight-labs` in it.

Reply:
Should I do that?